File: BroadcastBrain/Views/Components/Tokens.swift
import SwiftUI

extension Color {
    init(hex: String) {
        let s = hex.trimmingCharacters(in: .alphanumerics.inverted)
        var v: UInt64 = 0
        Scanner(string: s).scanHexInt64(&v)
        let r = Double((v & 0xFF0000) >> 16) / 255
        let g = Double((v & 0x00FF00) >> 8) / 255
        let b = Double(v & 0x0000FF) / 255
        self.init(red: r, green: g, blue: b)
    }

    static let bgBase      = Color(hex: "#050505")
    static let bgRaised    = Color(hex: "#0A0A0A")
    static let bgSubtle    = Color(hex: "#141414")
    static let bgHover     = Color(hex: "#171717")
    static let bbBorder    = Color(hex: "#262626")
    static let borderSoft  = Color(hex: "#1A1A1A")
    static let textPrimary = Color(hex: "#FAFAFA")
    static let textMuted   = Color(hex: "#A3A3A3")
    static let textSubtle  = Color(hex: "#737373")
    static let live        = Color(hex: "#EF4444")
    static let verified    = Color(hex: "#10B981")
    static let esoteric    = Color(hex: "#F59E0B")
    static let tactical    = Color(hex: "#F97316")
}

enum Typography {
    static let heroStat    = Font.system(size: 48, weight: .medium, design: .monospaced)
    static let statLabel   = Font.system(size: 14, weight: .regular, design: .monospaced)
    static let chip        = Font.system(size: 11, weight: .medium, design: .monospaced)
    static let body        = Font.system(size: 13, weight: .regular, design: .monospaced)
    static let playerName  = Font.system(size: 20, weight: .semibold, design: .monospaced)
    static let sectionHead = Font.system(size: 11, weight: .semibold, design: .monospaced)
}

File: app/src/theme/tokens.ts
// Design tokens — matches frontend/design-handoff/feature-1/frame.jsx TOKENS verbatim.
// Any colour/typography drift between this file and the handoff is a bug.

export const tokens = {
  bgBase:     '#050505',
  bgRaised:   '#0A0A0A',
  bgSubtle:   '#141414',
  bgHover:    '#171717',
  border:     '#262626',
  borderSoft: '#1A1A1A',
  text:       '#FAFAFA',
  textMuted:  '#A3A3A3',
  textSubtle: '#737373',
  live:       '#EF4444',    // red — hot, streak live, record in reach, pinned
  verified:   '#10B981',    // green — Sportradar source, saved style, achievement
  esoteric:   '#F59E0B',    // amber — tactical keyword, formation chip, AI nudge
  stickyBg:   '#F4E27A',    // commentator handwritten note
  stickyText: '#3D2D0A',
} as const;

export type Token = keyof typeof tokens;

// iPad landscape design canvas — matches handoff proportions.
export const IPAD = { WIDTH: 1366, HEIGHT: 1024 } as const;

// Font family string. Use the Google-Fonts-loaded name once loaded,
// with system monospace fallback strings that browsers understand.
export const FONT_MONO = 'IBMPlexMono';

// em-to-px letter-spacing helper. RN letterSpacing is numeric (px).
// Matches handoff values: tracking="0.14em" at fontSize 12 → 12 * 0.14 = 1.68.
export const tracking = (emFraction: number, fontSize: number) => emFraction * fontSize;
